src/Data/Text/Lazy.hs

```diff
@@ -1365,6 +1367,22 @@ break p t0 = break' t0
                    | otherwise -> let (a,b) = T.splitAt n t
                                   in (Chunk a Empty, Chunk b ts)
 
+-- | /O(n)/ Similar to 'break', but searches from the end of the string.
+--
+-- >>> T.breakEnd (=='0') "180cm"
+-- ("180","cm")
+breakEnd :: (Char -> Bool) -> Text -> (Text, Text)
+breakEnd p src = breakEnd' (reverseSpine src) where
+  reverseSpine = go Empty where
+    go res Empty        = res
+    go res (Chunk t ts) = go (Chunk t res) ts
+  breakEnd' = go Empty where
+    go r Empty = (empty, r)
+    go r (Chunk t ts) = case T.breakEnd p t of
+      ("", _) -> go (Chunk t r) ts
+      (l, r') -> (reverseSpine (Chunk l ts), Chunk r' r)
```

Comment:
Looks good, but please do not enable `OverloadedStrings` in this module. 

Would it be possible to implement via `foldrChunks`? Something like 
```haskell
breakEnd :: (Char -> Bool) -> Text -> (Text, Text)
breakEnd p = foldrChunks go (empty, empty) 
    where 
        go x (ys, zs) 
            | null ys   = let (y, z) = T.breakEnd p x in (chunk y empty, chunk z zs)
            | otherwise = (chunk x ys, zs)
```